packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1838,7 +2168,110 @@ class _TrackTapGestureRecognizer extends TapGestureRecognizer {
+  }) : _customPaintKey = customPaintKey,
+       _isLeadingButton = isLeadingButton,
+       super(
+         postAcceptSlopTolerance: postAcceptSlopTolerance,
+         supportedDevices: supportedDevices,
+         debugOwner: debugOwner,
+         duration: pressDuration,
+      );
+
+  final GlobalKey _customPaintKey;
+  final bool _isLeadingButton;
+
+  @override
+  bool isPointerAllowed(PointerDownEvent event) {
+    if (!_hitTestInteractive(_customPaintKey, event.position, event.kind)) {
+      return false;
+    }
+    return super.isPointerAllowed(event);
+  }
+
+  bool _hitTestInteractive(GlobalKey customPaintKey, Offset offset, PointerDeviceKind kind) {
+    if (customPaintKey.currentContext == null) {
+      return false;
+    }
+    final CustomPaint customPaint = customPaintKey.currentContext!.widget as CustomPaint;
+    final ScrollbarPainter painter = customPaint.foregroundPainter! as ScrollbarPainter;
+    final Offset localOffset = _getLocalOffset(customPaintKey, offset);
+    return painter.hitTestButtonsInteractive(localOffset, kind, _isLeadingButton);
+  }
+}
+
+// A long press gesture detector that responds to events on the leading scrollbar's button.
+class _LeadingButtonPressGestureRecognizer extends _ButtonPressGestureRecognizer {
+  _LeadingButtonPressGestureRecognizer({
+    double? postAcceptSlopTolerance,
+    Set<PointerDeviceKind>? supportedDevices,
+    required Object debugOwner,
+    required GlobalKey customPaintKey,
+    required Duration pressDuration,
+  }) : super(
+         true,
+         postAcceptSlopTolerance: postAcceptSlopTolerance,
+         supportedDevices: supportedDevices,
+         debugOwner: debugOwner,
+         customPaintKey: customPaintKey,
+         pressDuration: pressDuration,
+       );
+}
+
+// A long press gesture detector that responds to events on the trailing scrollbar's button.
+class _TrailingButtonPressGestureRecognizer extends _ButtonPressGestureRecognizer {
+  _TrailingButtonPressGestureRecognizer({
+    double? postAcceptSlopTolerance,
+    Set<PointerDeviceKind>? supportedDevices,
+    required Object debugOwner,
+    required GlobalKey customPaintKey,
+    required Duration pressDuration,
+  }) : super(
+    false,
+    postAcceptSlopTolerance: postAcceptSlopTolerance,
+    supportedDevices: supportedDevices,
+    debugOwner: debugOwner,
+    customPaintKey: customPaintKey,
+    pressDuration: pressDuration,
+  );
+}
+
 Offset _getLocalOffset(GlobalKey scrollbarPainterKey, Offset position) {
   final RenderBox renderBox = scrollbarPainterKey.currentContext!.findRenderObject()! as RenderBox;
   return renderBox.globalToLocal(position);
 }
+
+/// Base class for Scrollbar leading and trailing buttons shapes.
+///
+abstract class ScrollbarButtonsShape {
+  /// This abstract const constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const ScrollbarButtonsShape();
+
+  /// Return the button size.
```

Comment:
Are scrollbar buttons required to be square or is this just the extent of the button along the scrollbar's axis?

Small scrollbar buttons may have trouble with a11y and the [minimum tap target size](https://api.flutter.dev/flutter/material/MaterialTapTargetSize-class.html)

Reply:
+1 I think 48.0 is the minimum.